### ISSUE TITLE
Mathematical Instructions and Values (PI, E, Log, Abs, Root ...)

### ISSUE THREAD

**Issue by MarcosMD09** [colobot/colobot#1551]
Do you plan to add math instructions and constants to Colobot , such as ``PI``, ``E``, ``log``, ``abs``, ``root``, ...?

**Comment by tomaszkax86** [comment]
Yes, we'll add various functions and mathematical constants to Colobot. This could take some time because all of them will require adding documentation and translation files.

**Comment by MarcosMD09** [comment]
It's ok, it's just that I needed them for an advanced program.
<br>
<br>
By the way, here's an implementation of the `abs` statement:
```cpp
float Math::Abs(float value) {
    if(value < 0) {
        return value * -1;
    }

    return value;
}
```

**Comment by tomaszkax86** [comment]
No, there's a C++ function for calculating this without conditional instructions and you're using very slow method with multiplication.

**Comment by tomangelo2** [comment]
There already are some mathematical functions in game, like abs() or sqrt(), full list available in SatCom, in last section https://colobot.info/SatComOnline/master/cbot

It still lacks some functions, though.

**Comment by MarcosMD09** [comment]
> No, there's a C++ function for calculating this without conditional instructions and you're using very slow method with multiplication.

Oh, okay

**Comment by tomangelo2** [comment]
Was there a reason to close this issue? We could still add more mathematical functions/constants in future.

**Comment by MarcosMD09** [comment]
I committed a error

**Comment by MarcosMD09** [comment]
I committed a error.

**Comment by MarcosMD09** [comment]
Logarithmic functions `log` and `log10`:

```cpp
bool​ ​rLog(CBotVar* var, CBotVar* result, ​int​& ​exception​, ​void​* user) 
 ​{ 
 ​    ​float​   value; 
  
 ​    value = var->​GetValFloat​(); 
 ​    result->​SetSetValFloat(logf(value)); 
 ​    ​return​ ​true​; 
 ​}

bool​ ​rLog10(CBotVar* var, CBotVar* result, ​int​& ​exception​, ​void​* user) 
 ​{ 
 ​    ​float​   value; 
  
 ​    value = var->​GetValFloat​(); 
 ​    result->​SetSetValFloat(log10f(value)); 
 ​    ​return​ ​true​; 
 ​}
```

**Comment by MarcosMD09** [comment]
I suggested some functions, anyone can correct it?
